ledger-core/virtual/handlers/vcachedmemoryrequest.go

```diff
@@ -6,35 +6,131 @@
 package handlers
 
 import (
+	"context"
+
+	"github.com/insolar/assured-ledger/ledger-core/conveyor"
 	"github.com/insolar/assured-ledger/ledger-core/conveyor/smachine"
 	"github.com/insolar/assured-ledger/ledger-core/insolar/payload"
+	"github.com/insolar/assured-ledger/ledger-core/network/messagesender"
+	messageSenderAdapter "github.com/insolar/assured-ledger/ledger-core/network/messagesender/adapter"
 	"github.com/insolar/assured-ledger/ledger-core/vanilla/injector"
+	"github.com/insolar/assured-ledger/ledger-core/virtual/descriptor"
+	"github.com/insolar/assured-ledger/ledger-core/virtual/memorycache"
+	memoryCacheAdapter "github.com/insolar/assured-ledger/ledger-core/virtual/memorycache/adapter"
 )
 
 type SMVCachedMemoryRequest struct {
 	// input arguments
 	Meta    *payload.Meta
 	Payload *payload.VCachedMemoryRequest
+
+	messageSender messageSenderAdapter.MessageSender
+	memoryCache   memoryCacheAdapter.MemoryCache
+	pulseSlot     *conveyor.PulseSlot
+
+	object   descriptor.Object
+	response *payload.VCachedMemoryResponse
 }
 
 /* -------- Declaration ------------- */
+
 var dSMVCachedMemoryRequestInstance smachine.StateMachineDeclaration = &dSMVCachedMemoryRequest{}
 
 type dSMVCachedMemoryRequest struct {
 	smachine.StateMachineDeclTemplate
 }
 
-func (*dSMVCachedMemoryRequest) InjectDependencies(_ smachine.StateMachine, _ smachine.SlotLink, _ injector.DependencyInjector) {
+func (*dSMVCachedMemoryRequest) InjectDependencies(sm smachine.StateMachine, _ smachine.SlotLink, injector injector.DependencyInjector) {
+	s := sm.(*SMVCachedMemoryRequest)
+
+	injector.MustInject(&s.pulseSlot)
+	injector.MustInject(&s.messageSender)
+	injector.MustInject(&s.memoryCache)
 }
+
 func (*dSMVCachedMemoryRequest) GetInitStateFor(sm smachine.StateMachine) smachine.InitFunc {
 	s := sm.(*SMVCachedMemoryRequest)
 	return s.Init
 }
 
 /* -------- Instance ------------- */
+
+func (*SMVCachedMemoryRequest) InjectDependencies(sm smachine.StateMachine, _ smachine.SlotLink, injector injector.DependencyInjector) {
+	s := sm.(*SMVCachedMemoryRequest)
+
+	injector.MustInject(&s.pulseSlot)
+	injector.MustInject(&s.memoryCache)
+	injector.MustInject(&s.messageSender)
+}
+
 func (s *SMVCachedMemoryRequest) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
 	return dSMVCachedMemoryRequestInstance
 }
+
+func (s *SMVCachedMemoryRequest) migrationDefault(ctx smachine.MigrationContext) smachine.StateUpdate {
+	ctx.Log().Trace("stop processing VCachedMemoryRequest since pulse was changed")
+	return ctx.Stop()
```

Comment:
this is wrong migration, this request probably shouldn't have migration and continue working.

Reply:
removed